### ISSUE TITLE
Migrate to webpack built version on tests

### ISSUE THREAD

**Issue by saschanaz** [w3c/respec#2161]
Tests based on html files e.g. pubsubhub-spec.html still depend on requirejs. We may want to use webpack-built ReSpec instead to do #1975.

**Comment by saschanaz** [comment]
I'm not adding `good first issue` because I'm seeing a blocker: currently `makeRSDoc()` with webpack-built ReSpec finishes processing even before inserting our custom test configuration.

**Comment by devanshbatra04** [comment]
Just wanted some clarification. Are you referring to the requireJS usage [here](https://github.com/w3c/respec/blob/develop/tests/spec/core/pubsubhub-spec.html#L21)? Because pubsubhub-spec.html uses ```builds/respec-w3c-common.js``` (which as far as I can tell is the webpack build).
However this Require function comes from [core/expose-modules](https://github.com/w3c/respec/blob/develop/src/core/expose-modules.js) which isn't RequireJS (at least in case of pubsubhub)
 
Other HTML files ```webidl.html, highlight.html, includer.html and pre-process.html``` do use the RequireJS profile, I ran into some issue on [webidl.html](https://github.com/w3c/respec/blob/develop/tests/spec/core/webidl.html) but the others do work well by simply changing the script src.

**Comment by saschanaz** [comment]
I think you understand correctly. I think some other tests also had problems but I might be wrong.

**Comment by devanshbatra04** [comment]
WebIDL spec does have this problem. trying to come up with a solution.

**Comment by devanshbatra04** [comment]
I worked on this for several hours today.The issue encountered is only in ```core/webidl.html``` as of now.
I think I know exactly what is happening. 
When we fetch the dfn element form [definiton map here](https://github.com/w3c/respec/blob/develop/src/core/link-to-dfn.js#L103), In test environment with webpack build, we get an object that is identical to the dfn element originally pushed here. 

However this ```dfn``` object is not a DOM Element, and any operations on it do not show up in the DOM. 
This causes addId function [here](https://github.com/w3c/respec/blob/develop/src/core/link-to-dfn.js#L139) useless. As it attempts to set an ID on this static object copy of the required dfn element. Making it look like the tests executed before ```awaitRSDoc()``` was complete and the ID remained Unset. It causes the tests to fail which require dynamically setting of id attribute on the dfn elements.

**Comment by devanshbatra04** [comment]
This doesn't happen on running the HTML file in browser where the Map stores the actual HTMLElements. This is probably a bug.

**Comment by saschanaz** [comment]
>Importing DefinitionMap (in Webpack build on Karma) degrades HTMLElements to simple javascript objects

@devanshbatra04 That sounds bad, do you have an idea which exact line degrades them?

**Comment by devanshbatra04** [comment]
It works totally fine in the ```dfn-map``` when I tried extracting the dfn htmlElement [here](https://github.com/w3c/respec/blob/develop/src/core/dfn-map.js#L14)

However, [this](https://github.com/w3c/respec/blob/753f0803ae81d8370cb9dc557af4126d968db44b/src/core/link-to-dfn.js#L14) is where it gets messed up. And addID function [here](https://github.com/w3c/respec/blob/753f0803ae81d8370cb9dc557af4126d968db44b/src/core/link-to-dfn.js#L137) just doesn't have any effect. 

Basically the dfn objects that are fetched [here](https://github.com/w3c/respec/blob/753f0803ae81d8370cb9dc557af4126d968db44b/src/core/link-to-dfn.js#L103) are no longer the htmlElements that were pushed. They got degraded somehow to normal JS Objects. And setting ```Element.id``` wouldn't have effect.

**Comment by saschanaz** [comment]
When you try `document.contains(YOUR_EXTRACTED_ELEMENT)` what's the result?
